app/views/shared/_page_unload_warning.js.erb

```diff
@@ -0,0 +1,5 @@
+function pageUnloadWarning() {
+  return true;
+}
+
+window.onbeforeunload = pageUnloadWarning;
```

Comment:
Since this is pure JS, without any ERB, I believe it can go inside a `.js` file in the `assets/javascripts` directory, and then you can use `javascript_include_tag` inside each view where you want to include it, as opposed to using a helper method. That way, you won't have to nonce it since it won't be inline.